docs/dcp2_system_design.rst

```diff
@@ -986,6 +1065,400 @@ Broad Institute).
+   to the former.
+
+2. Update the entity describing the data file, as specified in `Update an
+   entity`_. Do so even if the entity document itself did not change. This is
+   necessary so that the ``entity_type``, ``entity_id`` and ``version`` fields
+   in the name of the descriptor object match those in the name of the
+   object containing the entity that describes the data file.
+
+
+Update a subgraph
+~~~~~~~~~~~~~~~~~
+
+Write the updated subgraph to an object in the ``links`` directory of the
+staging area. The name of the subgraph object must have the same ``links_id``
+and ``project_id`` as the original and a strictly higher ``version``.
+
+Note that it is uncommon for a subgraph to change without entities being added
+or removed. The scenarios during which it occurs naturally include the
+reordering of entities in the subgraph or migrating the subgraph document to a
+newer schema version.
+
+
+Add an entity
+~~~~~~~~~~~~~
+
+Write the entity document to an object in the ``metadata`` directory of the
+staging area. The ``entity_id`` in the name of the object must be different
+from the ``entity_id`` column of any row in any table of the TDR dataset the
+staging is imported into.
+
+This type of update creates an orphaned entity so it should only be used as
+part of other updates, as described in `Add a subgraph`_, `Add a project`_ or
+`Add an entity to existing subgraphs`_.
+
+
+Add a data file
+~~~~~~~~~~~~~~~
+
+1. Write the data file to an object in the ``data`` directory of the staging
+   area.
+
+2. Write the descriptor to an object in the ``descriptors`` directory of the
+   staging area. The ``file_uuid`` property of the descriptor must be
+   different from any ``file_uuid`` property of the JSON document in the
+   ``descriptor`` column of any row in any ``…_file`` table in the TDR dataset
+   the staging area will be imported into.
+
+   The ``entity_id`` and ``version`` fields in the name of the descriptor must
+   match those in the name of the object created in the next step.
+
+3. Add the entity describing the data file as defined in `Add an entity_`.
+
+This type of update creates an orphaned data file so it should only be used as
+part of other updates, as described in `Add a subgraph`_, `Add a project`_ or
+`Add an entity to existing subgraphs`_.
+
+
+Add a subgraph
+~~~~~~~~~~~~~~
+
+1. Add any entities introduced by the subgraph as described in `Add an
+   entity`_. If the subgraph also refers to entities that were previously
+   imported, those entities should not be written to the staging area.
+
+2. Write the subgraph to an object in the ``links`` directory of the staging
+   area. The ``links_id`` field in the name of that object must be different
+   from the value of the ``links_id`` column in any row of the ``links`` table
+   of the TDR dataset the staging are is imported into.
+
+
+Add a project
+~~~~~~~~~~~~~
+
+1. Add the ``project`` entity as described in `Add an entity`_
```

Comment:
what is meant by `project entity` here?

Reply:
An object whose path starts with `metadata/project_` containing a document that matches this schema:

https://github.com/HumanCellAtlas/metadata-schema/blob/master/json_schema/type/project/project.json

These are not present in the staging areas created by your adapters.

Reply:
Thanks for clarifying! Just wanted to make sure this was not something new we need to add